ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts

```diff
@@ -0,0 +1,148 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { BaseActivityPicklistQuestion } from './baseActivityPicklistQuestion.component';
+import { NGXTranslateService } from '../../../services/internationalization/ngxTranslate.service';
+import { Subject } from 'rxjs';
+import { FormControl } from '@angular/forms';
+import { debounceTime, distinctUntilChanged, map, takeUntil } from 'rxjs/operators';
+
+export interface Suggestion {
+    label: string;
+    isParent?: boolean;
+    parent?: string;
+    value: string;
+}
+
+@Component({
+    selector: 'ddp-activity-autocomplete-picklist-question',
+    template: `
+    <mat-form-field class="width ddp-autocomplete-field">
+        <input matInput
+               #autocompleteInput
+               [formControl]="inputFormControl"
+               [placeholder]="block.picklistLabel"
+               [matAutocomplete]="autoCompleteFromSource" />
+
+        <mat-autocomplete #autoCompleteFromSource="matAutocomplete" class="autoCompletePanel" [displayWith]="displayAutoComplete">
+            <mat-option *ngFor="let suggestion of filteredSuggestions"
+                        class="autoCompleteOption"
+                        [value]="suggestion"
+                        [class.parent]="suggestion.isParent"
+                        [class.child]="suggestion.parent">
+                <span [innerHtml]="suggestion.label | searchHighlight: autocompleteInput.value"></span>
+            </mat-option>
+        </mat-autocomplete>
+    </mat-form-field>`,
+    styles: [`
+        .width {
+            width: 100%;
+        }
+
+        .ddp-autocomplete-field ::ng-deep .mat-form-field-infix {
+          width: auto;
+        }
+
+        .parent {
+            font-weight: bold;
+        }
+
+        .child ::ng-deep .mat-option-text {
+            padding-left: 13px;
+        }
+    `]
+})
+export class AutocompleteActivityPicklistQuestion extends BaseActivityPicklistQuestion implements OnInit, OnDestroy {
+    private allSuggestions: Suggestion[] = [];
+    private readonly ngUnsubscribe = new Subject();
+
+    filteredSuggestions: Suggestion[];
+    inputFormControl = new FormControl();
+
+    constructor(translate: NGXTranslateService) {
+        super(translate);
+    }
+
+    public ngOnInit(): void {
+        this.allSuggestions = this.generateSuggestions();
+        const customVersionId = this.block.picklistOptions.find(option => option.allowDetails)?.stableId;
+
+        this.inputFormControl.valueChanges.pipe(
+            map(value => typeof value === 'string' ? value.trim() : value),
+            distinctUntilChanged(),
+            debounceTime(200),
+            takeUntil(this.ngUnsubscribe)
+        ).subscribe((value: string | Suggestion) => {
```

Comment:
Do we have to worry about unsubscribing?

Reply:
we do unsubscribe on the line above